crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -458,6 +472,80 @@ impl WritebackCache {
         )
     }
 
+    fn get_transaction_lock(
+        &self,
+        obj_ref: &ObjectRef,
+        epoch_store: &AuthorityPerEpochStore,
+    ) -> SuiResult<Option<LockDetails>> {
+        match self.locked_transactions.entry(*obj_ref) {
+            DashMapEntry::Vacant(vacant) => {
+                let tables = epoch_store.tables()?;
+                let lock = tables.get_locked_transaction(obj_ref)?;
+                if let Some(lock_details) = lock {
+                    vacant.insert(lock_details);
+                }
+                Ok(lock)
+            }
+            DashMapEntry::Occupied(occupied) => Ok(Some(*occupied.get())),
+        }
+    }
+
+    /// Attempts to atomically test-and-set a transaction lock on an object.
+    /// If the lock is already set, it will be overridden if the old lock is from a previous epoch.
+    /// If the lock is set successfully, the previous lock (if any) is returned
+    fn try_set_transaction_lock(
+        &self,
+        obj_ref: &ObjectRef,
+        new_lock: LockDetails,
+        epoch_store: &AuthorityPerEpochStore,
+    ) -> SuiResult {
+        // entry holds a lock on the dashmap shard, so this function operates atomicly
+        let entry = self.locked_transactions.entry(*obj_ref);
+
+        // TODO: currently, the common case for this code is that we will miss the cache
```

Comment:
Given this, and the fact that we are able to do multi-point DB read to get transaction locks, I am leaning towards getting all entries first (using sorted ids), find out which ones are missing, then do a multi-point read to get all of them.

Reply:
it is not very straightforward to do this - trying to acquire multiple entries into a dashmap will deadlock if any of them are on the same shard. So you'd have to use the raw api, get all the shards, de-dup them, lock the shards individually...